Test markhor

myriamlacroix opened the pull request:
- Remove markhor_estop in files (capra_estop).
- Fix include GPIO in estop.cpp.

myriamlacroix committed:
Fix include jetsonGPIO

myriamlacroix committed:
Fix launch file

lvanasse commented on src/estop.cpp:
```diff
@@ -1,17 +1,17 @@
 #include "ros/ros.h"
 #include "std_srvs/Trigger.h"
-#include "JetsonXavierGPIO/jetsonGPIO.h"
+#include "JetsonXavierGPIO/jetsonGPIO.c"
```
This is not what we want, this is very hacky, and I would not consider it correct. 

What we need to do is that in the JetsonXavierGPIU/jetsonGPIO.h we need to add define to allow for include in C++ code. 

Here's an explanation for it  : https://isocpp.org/wiki/faq/mixing-c-and-cpp#include-c-hdrs-personal

lvanasse replied:
Also, ideally, we should add abstraction to be able to have more portable code. So we could have a kind of BSP where we have `markhor_estop.h`, which would compile/include different type of header file depending on which robot we are building for.  But this might be out of scope for this PR

lvanasse replied:
Hum the document seems to say it's fine, so if you tell me that it works alright, I won't ask you to change it. 

http://wiki.ros.org/roscpp/Overview/Initialization%20and%20Shutdown

myriamlacroix replied:
It was tested, with .h it didn't work, so I copied the include from the old module markhor_estop, which was .c and worked. I  can still change it if needed.

lvanasse replied:
Yes, if you could change it that would be better, thank you.

lvanasse commented on src/estop.cpp:
```diff
@@ -34,20 +35,22 @@ bool toggleEstopDisable(std_srvs::Trigger::Request &req, std_srvs::Trigger::Resp
 }
 
-int main(int argc, char **argv) {
+int main(int argc, char **argv)
```
Also, was this tested in any way ? 

Looking up how to disable the e-stop once we want to shut down the node seems to require a little more code :

https://answers.ros.org/question/27655/what-is-the-correct-way-to-do-stuff-before-a-node-is-shutdown/

lvanasse commented on src/estop.cpp:
```diff
@@ -34,20 +35,22 @@ bool toggleEstopDisable(std_srvs::Trigger::Request &req, std_srvs::Trigger::Resp
 */
 void mySigintHandler(int sig)
```
This function name could be named better, I think

lvanasse reviewed (changes requested):
There's modification that needs to be done, specifically the `#include` for jetsonGPIO and I don't think that the feature of disabling the GPIO when shutting down really work. Was this tested ?

saxtot commented:
For the record, the code was tested and the node by itself worked as expected.

myriamlacroix committed:
Change function name.

lvanasse reviewed (commented):
Simply changing the .h instead of the .c and you should be good to go.

myriamlacroix committed:
Change include .c to .h

- Change include in estop.cpp
- Replace submodule JetsonXAvierGPIO to change file jetsonGPIO"h
